src/env_hooks.c

```diff
@@ -99,7 +99,7 @@ rpl_putenv(PUTENV_CONST char *string)
     len = (size_t)(equal - string) + 1;
     for (ep = environ; *ep != NULL; ep++) {
 	if (strncmp(string, *ep, len) == 0) {
-	    *ep = (char *)string;
```

Comment:
These casts are needed on systems where the argument to putenv() is const char *, like older Solaris and 4.4BSD.